cmd/utils/flags.go

```diff
@@ -1730,20 +1730,21 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 		// setEtherbase has been called above, configuring the miner address from command line flags.
 		if cfg.Miner.PendingFeeRecipient != (common.Address{}) {
 			developer = accounts.Account{Address: cfg.Miner.PendingFeeRecipient}
-		} else if accs := ks.Accounts(); len(accs) > 0 {
-			developer = ks.Accounts()[0]
 		} else {
-			developer, err = ks.NewAccount(passphrase)
-			if err != nil {
-				Fatalf("Failed to create developer account: %v", err)
+			if accs := ks.Accounts(); len(accs) > 0 {
+				developer = ks.Accounts()[0]
+				if err := ks.Unlock(developer, passphrase); err != nil {
+					Fatalf("Failed to unlock developer account: %v", err)
+				}
+			} else {
+				developer, err = ks.NewAccount(passphrase)
+				if err != nil {
+					Fatalf("Failed to create developer account: %v", err)
+				}
 			}
-		}
-		// Make sure the address is configured as fee recipient, otherwise
-		// the miner will fail to start.
-		cfg.Miner.PendingFeeRecipient = developer.Address
-
-		if err := ks.Unlock(developer, passphrase); err != nil {
```

Comment:
We need the developer account unlocked in the keystore so that users can send transactions from that account..